src/components/water.js

```diff
@@ -140,9 +141,7 @@ MobileWater.prototype = Object.create(THREE.Mesh.prototype);
 MobileWater.prototype.constructor = THREE.Water;
 
 async function loadWaterNormals(url) {
-  const texture = await new Promise((resolve, reject) =>
-    new THREE.TextureLoader().load(url, resolve, undefined, reject)
-  );
+  const texture = await new Promise((resolve, reject) => new HubsTextureLoader().load(url, resolve, undefined, reject));
```

Comment:
Out of scope for this PR, but its a bit annoying how we have this promise code everywhere. Would be nice to just add a function to HubsTextureLoader like `loadAsync` that just wraps `load` in a promise for you. (might not be a bad idea for the loader API updstream in general honestly)